### PR TITLE
Allow removal of item definition fields

### LINKED ISSUE #14644: Removing Variables from Item Def/Definitions
### Problem

Sorry if this is a duplicate or isn't formatted properly. I looked for a tiny bit to see if there was a thread opened by appgurueu but I couldn't find one, and this is my first time using GitHub

A thread I opened a few weeks ago https://forum.minetest.net/viewtopic.php?t=30405 went into detail about how to remove functions from an Item's definition. Was told I should open an issue about it by LMD aka @appgurueu :D 

There is no official engine function or way (to my knowledge) to remove an unwanted variable - such as "on_use" in this case. Currently I set it to false as a method of committing to this - but this isn't an officially supported way of doing such.

### Solutions

Two solutions were proposed by appgurueu and with comments by @Montandalar (BlockHead)

appgurueu's solutions:
> I think we should have a way to let override_item delete fields, that definitely seems like a legitimate feature to me. One way to implement this could be using a special, unique minetest.nil_placeholder (could be a userdatum) which could be used instead of false (the problem with false is that it does not work for fields which support booleans; for other fields it's fine however). Or adding another parameter with a list of fields to be deleted?

Montandalar's comment;
> I think I like the delete-list better because it extends the API with an extra argument - of course, there may be weird code out there using an extra argument already, because you should expect the unexpected, but that's not really a supported use case. I like the delete-list because doesn't have a magic userdata* value which is probably what would be needed, and also would not be present in older versions whereas the delete-list would pass through and basically be ignored. I suppose that makes it a question of crash vs no crash for a bit of code that doesn't check for compatibility of what it is about to use.

The probably best solution would be to go with the second solution that appgurueu suggested: another parameter - a table of indexes to be removed. But I imagine both solutions could work out just as well?

### Alternatives

Well as I mention in the problem, I currently use "false" as a way to modify a function variable so that it is not used.

### Additional context

Just be sure to check https://forum.minetest.net/viewtopic.php?t=30405 for anything extra :D

### PR DESCRIPTION
**Pull request by appgurueu**
Closes #14644

### CODE REVIEW

**Comment by TubberPupper** [comment]
Huh, that's cool, I never heard of rawset :O

File: builtin/game/register.lua [hunk 1]
```diff
@@ -418,6 +418,9 @@ function core.override_item(name, redefinition)
 	for k, v in pairs(redefinition) do
 		rawset(item, k, v)
 	end
+	for _, field in ipairs(del_fields) do
```
Comment (sfan5) [inline builtin/game/register.lua:421]: needs to be optional

Comment (TubberPupper) [reply builtin/game/register.lua:421]: Oh yeah, that'd cause a crash if ran without

Run a type(del_fields) == "table" ?

If you ran with a if del_fields then wouldn't that cause an error if it was given say, a number or boolean? ;p

**Comment by sfan5** [comment]
Ah, I forgot.
This needs an entry in `core.features`.

**Comment by sfan5** [comment]
Also I guess an unit test would be nice (and easy to write).

**Comment by appgurueu** [comment]
I don't think there's much point in a feature flag for this, but it surely won't hurt. Pushing [this](https://gist.github.com/appgurueu/65bdced10db37e4b9a07ede57ec7a989) soon.

**Pull request merged**